File: baekjoon/단계별로 풀어보기/스택/1874_스택 수열.py
# https://www.acmicpc.net/problem/1874

from sys import stdin

input = stdin.readline
stack = []
label = []
answer = []
n = int(input())
for _ in range(n):
    label.append(int(input()))
k = label.pop(0)
for i in range(1, n + 1):
    stack.append(i)
    answer.append('+')
    while stack and stack[-1] == k:
        stack.pop(-1)
        answer.append('-')
        if label:
            k = label.pop(0)
if stack:
    print('NO')
else:
    print('\n'.join(answer))

File: C#/Baekjoon/스택/1874_스택 수열.cs
//https://www.acmicpc.net/problem/1874

using System.Text;

using var sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
var sb = new StringBuilder();
var s = new Stack<int>();
int n = int.Parse(sr.ReadLine()), m = n, input, i = 1, last;
while (n-- > 0)
{
    input = int.Parse(sr.ReadLine());
    while (i <= m + 1)
    {
        
        if (s.TryPeek(out last) && last == input)
        {
            s.Pop();
            sb.AppendLine("-");
            break;
        }
        else
        { 
            s.Push(i); 
            sb.AppendLine("+");
            i++;
        }
    }
}
Console.WriteLine(s.Count == 0 ? sb : "NO");